### DISCUSSION TITLE
Why does my device not update automatically?

### DISCUSSION THREAD

**Discussion by tsutsuku** [tuya/tuya-home-assistant#113] [Q&A]
Check if "Device status notification" subscribed.

<img width="1160" alt="WeChatWorkScreenshot_181aed4a-7fea-4f9f-9358-501a66796ba3" src="https://user-images.githubusercontent.com/18442047/124885528-f1bbb100-e005-11eb-92f5-f34ce80db98f.png">

**Comment by MickW69** [comment]
Does this mean I need to setup mqtt to listen for the status?

**Reply by tsutsuku** [reply]
No, tuya v2 integration already setup it. 
You need to subscribe in tuya iot platform.

**Comment by sebadv** [comment]
Could you please provide the URL for this page? I can't seem to find it on de IoT site.

**Comment by ppmt** [comment]
I eventually found out how to activate the notification. 

Go to Cloud/My Services/All Services and search for "device status notification"
You can then subscribe to it and add your project to it.

I had to restart HA and now it works perfectly....

Update are extremely fast.

**Reply by sebadv** [reply]
Fantastic, Thanks! Updates are indeed very fast. They appear at the same speed of the Smartlife/Tuya app.

**Reply by ppmt** [reply]
Yes but when I unplug my device (not battery powered) then the integration stops receiving data even after I plug it back
I have to restart HA to recover from it

**Reply by ravac1** [reply]
In my case, I have the device status notifications enabled and the update still does not work. I even turn lights on from HA and the status does not change. 

Later edit: my case is different, I opened separate issue for it

**Reply by tsutsuku** [reply]
Any logs with tuya-openmq?

**Reply by ppmt** [reply]
I don't know what this tuya-opnemq is so can't provide logs. Is that a program? or something that needs to be activated?
I have activated the log in Home Assistant as per the requirement but don't know what else to do!

**Comment by Tristran** [comment]
I made the mistake of not adding the device status notificaiton to my project when I first created it. I've since added and authorised it but can't get notifications working. My devices refresh status once after home assistant restarts and then the devices "become unavailable". I've reinstalled the tuya v2 integration and no change.

**Reply by tsutsuku** [reply]
You can create a issue with your device specification.

**Comment by rodgrech** [comment]
i have checked and triple checked, i have it subscribed to the service
![image](https://user-images.githubusercontent.com/77467195/128992017-4e4cd6fa-5f8b-4679-87af-8ac6910e7eae.png)

**Comment by MustardtheCalico** [comment]
This works fine with my Tuya Wifi device (plug), but appears not to work for my Zigbee based tuya devices (Lidl lamps/plug setup via the tuya app) - Can see and control them on/off but it doesn't remember the status. 

Is there another setting somewhere specific to Zigbee devices compared to wifi?

**Comment by creator72** [comment]
I have the same problem as @gavinsmith1971  the integration updates only on restart of HA, I have a BLE tuya gateway, I can set the temperature of my TRV's and turn them off and on, I see the updates in the smartlife app but HA does not update until I restart it. I have the notification API enabled but its made no difference.  The "History" page in HA shows "Device became unavailable 10 minuets ago" and never updates

**Comment by indiocochino** [comment]
Same for me...
No update in Home Assistant.
Even if I change the status through the device debuggin option in iot.tuya...it doesn't change the status in Home Assistant.
Could you please try to help?

**Comment by jypsie** [comment]
can install v2, but I do not get any device in my integration..... old Tuya works fine

**Reply by saneone** [reply]
Same for me... I cannot see any devices in HA with Tuya V2 integration, while the old integration works.

**Reply by jypsie** [reply]
Well I got it to the point I got all devices in that IOT cloud and a part in HA. They could not have made it more complicated so I can not explain anymore what I did. At the end I lost 90% of my functionality in HA and the 10% remaining don't even work reliable......

**Comment by salreus1** [comment]
I am having the same issue. I am using ver 1.5. HA loads all the sensors but there is not real time updates. I do get an update if I restart HA so I know it queries at least 1 time upon restart but there is no live updates. I am using Beelite US Smart Wifi Plug Smart Socket.

**Reply by salreus1** [reply]
I solved the issue by going into IOT.TUYA and starting over and making sure I added "device status notification" at the start of the project as see in this vid. https://youtu.be/f8-7Hvrmh3Y?t=537 . Once I authenticated HA to the new project. I am getting updates.

**Comment by staszekj** [comment]
I have the same issue as above.
I found
"Device Status Notification" service is OK but
"IoT Core" service is expired. Maybe I'm wrong but I think "IoT Core" service is required because there is info on iot.tuya.com:

"Please make sure that there is a **basic resource pack** available under your account. You can subscribe to the **IoT core** service to get a free **basic resource pack**."

I clicked button "Extend Trial" and now my "... application for extension is being reviewed" by Tuya.
So we will see Tuya agree to extend my trial "IoT Core" service ;-)

**Reply by staszekj** [reply]
My IoT Core subscription was extended but it didn't help :-(
Maybe it works a little better better but delays are huge.

Eventually I decided to Local Tuya HA integration (more work to connect but finally works like a charm :-) ... and what was surprised me it works nice with Tuya Smart mobile app (both directions)

**Reply by svscorp** [reply]
@staszekj can you share a bit more details on the Local Tuya HA integration?

**Comment by ppmt** [comment]
I think I am going to give up on Tuya. My sensor works well for a bit of time after a HA restart but eventually stopped updating.

**Comment by cryogenx** [comment]
same issue no updates for device states... super annoying..

**Comment by zlinoliver** [comment]
Have you tried 2021.12.6 or above version of Home Assistant? This PR fix the device state update issue: https://github.com/home-assistant/core/pull/62858

**Reply by svscorp** [reply]
It's not fixed in latest version.

**Reply by plandregan** [reply]
Its now september, still not getting reliable refresh of device entity. So the power is pretty much useless.

Wierd thing is, if i open the Tuya App and look at the device power, the entity in HA suddently begins to update again
